### PR TITLE
Remove HARD_LIMIT when expanding wildcard queries.

### PR DESCRIPTION
**Pull request by bsura**

### CODE REVIEW

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/DiscoveryService.java [hunk 1]
```diff
@@ -42,6 +44,24 @@
...
+	 *  
+	 *  Fixing the max QPM (Queries Per Minute) for Argus to around 500, and avg. query latency to around 3 secs (both are conservative estimates), we would be serving 25 queries concurrently. Again these are just ball park
+	 *  estimates to fix the maximum number of datapoints that should be returned in a response.
+	 *  
+	 *  Let's assume we reserve 2GB of memory for concurrently executing these 25 queries. That would mean around 80MB per request. Roughly translating to around (80M bytes/40 bytes =) 2M datapoints. 
+	 *  
+	 *  Please configure this no. according to the above calculation for your environment. 
+	 **/
+	static final int MAX_DATAPOINTS_PER_RESPONSE = 2000000;
+    static final long DATAPOINT_SAMPLING_FREQ_IN_MILLIS = 60 * 1000L;
```
Comment (dilipdevaraj-sfdc) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/DiscoveryService.java]: Add comment that Argus least scale resolution of datapoints is 1 min, hence we are using that for DATAPOINT_SAMPLING_FREQ_IN_MILLIS

Comment (bsura) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/DiscoveryService.java]: Sure. Will add.

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/DiscoveryService.java [hunk 2]
```diff
@@ -92,6 +112,65 @@
...
+    	long timeWindowInMillis = query.getEndTimestamp() - query.getStartTimestamp();
+    	long downsamplingDivisor = (query.getDownsamplingPeriod() == null || query.getDownsamplingPeriod() <= 0) ? 1 : query.getDownsamplingPeriod(); 
+    	long numDatapointsPerTimeSeries = timeWindowInMillis / DATAPOINT_SAMPLING_FREQ_IN_MILLIS / downsamplingDivisor;
+    	
+    	numDatapointsPerTimeSeries = numDatapointsPerTimeSeries <= 0 ? 1 : numDatapointsPerTimeSeries; 
+    	
+    	return (int) (MAX_DATAPOINTS_PER_RESPONSE / numDatapointsPerTimeSeries);
+    }
+    
+    static int numTimeseriesForQuery(MetricQuery mq) {
```
Comment (dilipdevaraj-sfdc) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/DiscoveryService.java]: These two methods have slightly similar and confusing names
You can maybe change these to
getNumTimeseriesAllowed()
getNumTimeseriesExpandedTo()

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/DefaultDiscoveryService.java [hunk 3]
```diff
@@ -130,18 +125,25 @@ public DefaultDiscoveryService(SchemaService schemaService, SystemConfiguration
...
+            }
+            
             if (query.getTags() == null || query.getTags().isEmpty()) {
                 MetricSchemaRecordQuery schemaQuery = new MetricSchemaRecordQuery(query.getNamespace(), 
                 		query.getScope(), query.getMetric(), "*", "*");
                 int page = 1;
 
                 while (true) {
-                    List<MetricSchemaRecord> records = _schemaService.get(schemaQuery, 200, page++);
+                    List<MetricSchemaRecord> records = _schemaService.get(schemaQuery, 500, page++);
```
Comment (dilipdevaraj-sfdc) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/DefaultDiscoveryService.java]: Please use static final int MAX_QUERIES_PER_MIN = 500

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/DefaultDiscoveryService.java [hunk 4]
```diff
@@ -164,18 +165,23 @@ public DefaultDiscoveryService(SchemaService schemaService, SystemConfiguration
...
             } else {
+            	Map<String, Integer> timeseriesCount = new HashMap<>();
                 for (Entry<String, String> tag : query.getTags().entrySet()) {
                     MetricSchemaRecordQuery schemaQuery = new MetricSchemaRecordQuery(query.getNamespace(), query.getScope(), query.getMetric(),
                         tag.getKey(), tag.getValue());
                     int page = 1;
 
                     while (true) {
-                        List<MetricSchemaRecord> records = _schemaService.get(schemaQuery, 200, page++);
+                        List<MetricSchemaRecord> records = _schemaService.get(schemaQuery, 500, page++);
```
Comment (dilipdevaraj-sfdc) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/DefaultDiscoveryService.java]: Here too

**Comment by bsura** [comment]
I will merge this PR and make the required changes as part of another PR.

**Pull request merged**